packages/material-ui/src/ClickAwayListener/ClickAwayListener.js

```diff
@@ -5,82 +5,92 @@ import ReactDOM from 'react-dom';
 import PropTypes from 'prop-types';
 import EventListener from 'react-event-listener';
 import ownerDocument from '../utils/ownerDocument';
+import { useForkRef } from '../utils/reactHelpers';
+
+function useMountedRef() {
+  const mountedRef = React.useRef(false);
+  React.useEffect(() => {
+    mountedRef.current = true;
+    return () => {
+      mountedRef.current = false;
+    };
+  }, []);
+
+  return mountedRef;
+}
 
 /**
  * Listen for click events that occur somewhere in the document, outside of the element itself.
  * For instance, if you need to hide a menu when people click anywhere else on your page.
  */
-class ClickAwayListener extends React.Component {
-  mounted = false;
-
-  moved = false;
-
-  componentDidMount() {
-    // Finds the first child when a component returns a fragment.
-    // https://github.com/facebook/react/blob/036ae3c6e2f056adffc31dfb78d1b6f0c63272f0/packages/react-dom/src/__tests__/ReactDOMFiber-test.js#L105
-    this.node = ReactDOM.findDOMNode(this);
-    this.mounted = true;
+function ClickAwayListener(props) {
+  const { children, mouseEvent, touchEvent, onClickAway, ...other } = props;
+  const mountedRef = useMountedRef();
+  const movedRef = React.useRef(false);
+
+  const nodeRef = React.useRef();
+  // can be removed once we drop support for non ref forwarding class components
+  const handleOwnRef = React.useCallback(ref => {
+    nodeRef.current = ReactDOM.findDOMNode(ref);
+  }, []);
+  const handleRef = useForkRef(children.ref, handleOwnRef);
+
+  const handleClickAway = React.useCallback(
+    event => {
+      // Ignore events that have been `event.preventDefault()` marked.
+      if (event.defaultPrevented) {
+        return;
+      }
+
+      // IE 11 support, which trigger the handleClickAway even after the unbind
+      if (!mountedRef.current) {
+        return;
+      }
+
+      // Do not act if user performed touchmove
+      if (movedRef.current) {
+        movedRef.current = false;
+        return;
+      }
+
+      const { current: node } = nodeRef;
+      // The child might render null.
+      if (!node) {
+        return;
+      }
+
+      const doc = ownerDocument(node);
+
+      if (
+        doc.documentElement &&
+        doc.documentElement.contains(event.target) &&
+        !node.contains(event.target)
+      ) {
+        onClickAway(event);
+      }
+    },
+    [mountedRef, onClickAway],
+  );
+
+  function handleTouchMove() {
```

Comment:
I think that we need useCallback here too for https://github.com/oliviertassinari/react-event-listener/blob/master/src/index.js#L73.

Reply:
Right forgot about that one.

Reply:
@oliviertassinari Thanks for pushing forward!